stetho/src/main/java/com/facebook/stetho/inspector/network/DownloadingAsyncPrettyPrinterFactory.java

```diff
@@ -157,10 +158,18 @@ public Request(URL url) {
 
     @Override
     public String call() throws IOException {
-      InputStream urlStream = url.openStream();
-      String result = Util.readAsUTF8(urlStream);
-      urlStream.close();
-      return result;
+      HttpURLConnection.setFollowRedirects(false);
+      HttpURLConnection connection = (HttpURLConnection)url.openConnection();
+      int statusCode = connection.getResponseCode();
+      if (statusCode != 200) {
+        throw new IOException("Got status code: " + statusCode + " while downloading schema");
```

Comment:
The IOException will be caught when pretty printer's printTo method is trying to call "response.get()" to retrieve schema. If an IOException is thrown from here, printTo will catch it and print out the error message following with the original response body in the Response tab of Network.